Shared/Samples/Navigate route/NavigateRouteView.swift

```diff
@@ -0,0 +1,352 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import AVFoundation
+import SwiftUI
+
+struct NavigateRouteView: View {
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        MapView(
+            map: model.map,
+            viewpoint: model.viewpoint,
+            graphicsOverlays: model.graphicsOverlays
+        )
+        .onViewpointChanged(kind: .centerAndScale) { model.viewpoint = $0 }
+        .locationDisplay(model.locationDisplay)
+        .alert(
+            isPresented: Binding(get: { model.error != nil }, set: { _ in model.error = nil }),
+            presentingError: model.error
+        )
+        .overlay(alignment: .top) {
+            Text(model.statusText)
+                .frame(maxWidth: .infinity, alignment: .leading)
+                .padding(8)
+                .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+        }
+        .task {
+            // Solves the route and sets the navigation.
+            await model.solveRoute()
+            model.setNavigation()
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                Button("Navigate") {
+                    model.isNavigatingRoute = true
+                }
+                .task(id: model.isNavigatingRoute) {
+                    guard model.isNavigatingRoute else { return }
+                    await model.startNavigation()
+                }
+                .disabled(model.isNavigateDisabled || model.isNavigatingRoute)
+                
+                Spacer()
+                
+                Button("Recenter") {
+                    model.autoPanMode = .navigation
+                }
+                .disabled(!model.isNavigatingRoute || model.autoPanMode == .navigation)
+                
+                Spacer()
+                
+                Button("Reset") {
```

Comment:
I think it may have something to do with the viewpoint being set in WGS84 from the route geometry and it coming back out at as web mercator in `onViewpointChanged` but I'd have to debug further. Is it a blocker for this sample?

Reply:
I dug some more; on line 248 the viewpoint is created without a scale (`viewpoint = Viewpoint(boundingGeometry: routeGeometry, rotation: 0)`. If a scale is passed, (e.g. `viewpoint = Viewpoint(center: routeGeometry.extent.center, scale: 34182, rotation: 0)`) then the provided scale is preserved when `onViewpointChanged` fires.

Reply:
It doesn't block this sample. I think it might be worth logging an issue to take a further look though.

Reply:
`swift/issues/3854`